build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -414,4 +426,41 @@ public boolean visit(IResourceProxy proxy) throws CoreException {
 
 	}
 
+	private String findCompilerInPath(ITool tool, IConfiguration config) {
+		if (toolMap.containsKey(tool)) {
+			return "\"" + toolMap.get(tool) + "\""; //$NON-NLS-1$//$NON-NLS-2$
+		}
+		String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
+		File pathToCompiler = new File(compilerName);
+
+		if (pathToCompiler.isAbsolute()) {
+			toolMap.put(tool, compilerName);
+			return "\"" + compilerName + "\""; //$NON-NLS-1$ //$NON-NLS-2$
+		}
+		ICConfigurationDescription cfg = ManagedBuildManager.getDescriptionForConfiguration(config);
+		IEnvironmentVariable[] variables = CCorePlugin.getDefault().getBuildEnvironmentManager().getVariables(cfg,
+				true);
+
+		for (IEnvironmentVariable variable : variables) {
+			if ("PATH".equals(variable.getName())) { //$NON-NLS-1$
+				IPath resolvedPath = PathUtil.findProgramLocation(compilerName, variable.getValue());
+				if (resolvedPath != null) {
+					String path = resolvedPath.toString().replaceAll("\\s+", " ").trim(); //$NON-NLS-1$//$NON-NLS-2$
+					toolMap.put(tool, path);
+					return "\"" + path + "\""; //$NON-NLS-1$ //$NON-NLS-2$
+				} else {
+					return null; // Only one PATH so can exit early
+				}
+			}
+		}
+
+		return null;
+	}
+
+	private static String getCompilerName(ITool tool) {
+		String compilerCommand = tool.getToolCommand();
+		String[] arguments = CommandLineUtil.argumentsToArray(compilerCommand);
+		return arguments[0];
```

Comment:
Possible index out of bounds exception here. Generally that would only happen if there is a problem in the tool, but we should avoid the runtime exception here.